compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2870,6 +2870,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             );
                         }
                     }
+
+                    if let ty::RawPtr(..) = base_t.kind() && idx_t.is_integral() {
+                        err.span_help(
```

Comment:
I was hesistant to make them suggestions because `ptr::add` and friends are unsafe functions.

If I were to make it into a suggestion, it would become something like

```
unsafe { ptr1.add(5) }
```

Unless the suggestion doesn't add the unsafe block?

Reply:
`wrapping_add` isn't unsafe. Although C pointer arithmetic does have the same preconditions as `add` I think. But I don't think suggestions should include unsafe code. `add` should still be mentioned in the help though.

Reply:
> But I don't think suggestions should include unsafe code.

I think I agree with this. I can undo the change to make them suggestions and revert them to being help notes if this is deemed more suitable.

Reply:
You can have both a suggestion _and_ a note